### ISSUE TITLE
EasyList blocking whole website powered by Firework video

### ISSUE THREAD

**Issue by SyedSan1** [easylist/easylist#9588]
We, Firework, are a video platform company that businesses such as fluff, Safeway, Globo, Albertsons, Buzzfeed use to add short videos on their  website. We think EasyList is wrongly categorizing content powered by Firework as an ad, resulting in sever blocking and a very bad user experience. We enable native video content for publishers, e-commerce etc., and these are not advertisements. 

### List Subscriptions you're using:

AdBlock Plus, AdBlock, NoScript - I tried one by one to confirm these are blocking content.

### Your settings
This is happening in reportedly all browsers. 
I tested on Chrome (Windows 12) 
Samsung Internet (Android).

**Comment by ryanbr** [comment]
Example url where its broken, and as a community we'll all review whether should still be covered.

**Comment by SyedSan1** [comment]
Example url:
https://www.globo.com/dropz/
https://www.safeway.com/shop/lp/fresh-produce.html
https://fluff.co/

All of these websites are getting broken thru EasyList filter.
Globo with AdBlock Plus turned on gets completely blocked.
Safeway has implemented a video row, so that whole row turns blank because of the adblock.
And Fluff has a floating video at the bottom right that gets blocked.

**Comment by ryanbr** [comment]
The video ads seen (previously seen on https://www.playstationlifestyle.net/)  are obnoxious:

* Launched without user intervention.
* Video would autoplay.
* Video would scroll down.
* Would load the browser with additional 30-40 Network requests just from firework, WTF. 

![obnoxious](https://user-images.githubusercontent.com/1659004/140258824-95dce9c6-5372-4a00-b3ca-cd3452c686ea.png)

This was well within the reason for to be blocked. Like the examples where firework content is ads:

https://esportnow.pl/
https://www.khaboronline.com/
https://ahmedabadmirror.com/
https://esportway.com/aimbot-command-in-csgo/

Can't have it both ways.  If used for any non-ad purposes my vote would just make a exemption for that specific site.

**Comment by pgl** [comment]
@SyedSan1 

Hey there. Just a quick comment after having a look through a recent blog post (https://firework.tv/uk/blog/firework-and-internet-3-0/).

Everything there describes Firework as an ad and tracking platform.

> So, what are the soul searching questions driving the Firework, partner-first value systems outlined above?
> * When was the last time you visited a website?
> * When was the last time you clicked on an ad?
> * When was the last time you spent more than 20 seconds on a web page?

...

> Life and thriving in the future-state digital open web requires:
> * Leveraging native, content-born video pre-roll, interstitials, and branded sponsorship experiences versus intrusive ad units which track users

etc.

Also, the examples you gave, they're actually still just ads, right? Just promoting stuff for the site they're on?

**Comment by SyedSan1** [comment]
Hi ryanbr,
Thank you for your response.

The example you shared will not happen again. This was by mistake. All ad content will be served by Google Ad Manager that you can block at request level. But, the current implimentation is blocking entire video content, which is really bad user exp.
For example: 
![image](https://user-images.githubusercontent.com/93693440/140431406-db69c0d3-ff27-4605-ba76-185394d67f6c.png)

We are giving our player (just like jw player) to manage video and potentially monetize their website. The player and content should not be blocked while ads could be blocked.


In many ways, our floating video (in case of fluff.co) is similar to chat bots. It's deployed in partnership with the website publisher to engage their users and make websites more similar to social-media style video experience. 
The key brands such as Heinz, Albertsons won't deploy us if it were a bad user experience. a blanket ban on floating video doesn't make sense. We think the EasyList policy should also evolve with time. 

@pgl 
"Everything there describes Firework as an ad and tracking platform."
We are a video first saas company. The website publishers subscribe to us, and post video content. And they can choose to monetize content by inserting ads from Google ad manager. 

"Also, the examples you gave, they're actually still just ads, right? Just promoting stuff for the site they're on?
Every website promotes itself. Firework gives websites tools to promote their products. Blocking the content such as the example I shared above doesn't make sense as it results in bad user experience. 

For example: users go to watch videos on YouTube. You don't block entire video, but you can block youtube pre-roll ad.

**Comment by spirillen** [comment]
Is it only me who reads advertisers and "publishers" on there frontpage??

![image](https://user-images.githubusercontent.com/44526987/140443711-e6003f97-de45-4c1c-95a2-031d0959fdf2.png)

Can never know, I'm getting older... and the eyes is not what there used to be

**Comment by ryanbr** [comment]
Easylist policy has evolved over time to include invideo native ads. Saving bandwidth, browser resources. Compared to JWplayer (as an example), Firework seems to much heavy on resources given the amount of network requests being issued. Just some talking points:

- "It won't happen again" while its still happening on the sample sites provided in the previous post. Are these not ads? Essentially they're just banner ads with embedded videos.

- We also block some jw player embedded posts also. But because the vast majority of jwplayer is serving a purpose, while some minor domains just use it to serve video ads. It was make specific.  If jwplayer became more of an video ad platform, we could do the same with jwplayer and block it outright.

- Often the more annoying the ad the better, gathers more user attention. But we don't rate Ads on "Annoyance", we have policys to avoid confusion, and outline what is covered.

- fluff.co autoplays the video (and scrolls with the user), most chatbots will just sit there until the user interacts with it. Also including the other 3rd-party domains it loads fireworkadservices.com and fireworkanalytics.com.  Might treated differently if it didn't load so much javascript, onclick to load the video and didn't autoplay.

- A webpage dedicated to firework videos (as seen on globo.com/dropz) is vastly different. If hypothetically globo decided to implement fireworkad videos all over the site. That would take it out of the single page sandbox and make it advert, which should be blocked.

As a user browsing a page, with a popup video ad window (or a banner with a video in it). As an adblock user, wouldn't you think should be covered, and why special treatment should be given.  Because no doubt, we would get user complaints of video ads showing up.

**Comment by jinchen-fw** [comment]
hi @ryanbr 

1. can you please be specific on "Firework seems to much heavy on resources given the amount of network requests being issued."?
2. "its still happening on the sample sites provided in the previous post. Are these not ads?" which videos you are referring to?  
3. Our product is also serving a purpose, to give a way to business and publisher to share information to their users using short form video. e.g, on safeway.com, users can watch video how a fruit shaved ice is being made. On fluff.co, users can watch the testimonial videos on fluff.co, only who are interested in buying a fluff product will be going to their website. And there is a prominent "x" button that user can close the video. 

we are offering a platform that customer can better connect with their user via short form video, they can choose to run ads in between those video but all ads are being served by ad server like Google Ad Manager, so it is already being blocked by adblockers at the VAST request level, so users will not see the ad, but there is no reason our player should be blocked.

**Comment by ryanbr** [comment]
Just loading https://esportnow.pl/ as an example:  https://i.imgur.com/CfdUKUJ.png  By just blocking one firework filter it prevented this. Saving my browser, and bandwidth avoided all of these network requests.

These ads: 
https://esportnow.pl/
https://www.khaboronline.com/
https://ahmedabadmirror.com/
https://esportway.com/aimbot-command-in-csgo/

I still saw these ads without any changes, so on a "VAST request level", it didn't matter for me.   Unless another filter list author can argue something differently, with a compelling reason to not block this. This is what an Adblocker is designed to block, and should block.

**Comment by jinchen-fw** [comment]
@ryanbr most of those network requests are for analytic purpose which you will see the same amount if not more when going to a website who embeds a youtube video on their page. You will not block the youtube video, don't you?  We are explicitly using fireworkanalytics.com so everyone knows that is for analytic purpose. You can add the domain fireworkadservices.com to your list. We have no objection on that. 
Back to the example of https://esportnow.pl/, when you load the site,  you will be given a preroll ad which is being inserted by IMA SDK which is already being blocked by most of the adblocker. After the real ad is ended, user can start consuming other contents, which are introducing new games, they are same as an article being post to introduce a new game. Users who is going to esprotnow.pl to consume those content. In our humble opinion, those are not ads and users do engage with those videos.

**Comment by SyedSan1** [comment]
@spirillen  
"Is it only me who reads advertisers and "publishers" on there frontpage??"

We are not saying we don't enable publishers and advertisers with our tool. Our tool enables both. 
Note, we are not asking to unblock ads - it's fair to block ads. We only want to get the native video content to play on the website because our video enhances the publisher website. 
For example: look at this beautiful video Safeway has implemented showing Thanksgiving recipe - scrolldown a little on safeway.com

![image](https://user-images.githubusercontent.com/93693440/142566764-5f7cf5ed-5bf0-4a41-bfdc-d32b0b06519f.png)

The blanket blocking is causing huge problem for some publishers and they are reaching out for help.

**Comment by spirillen** [comment]
Hey @SyedSan1 

I have no more power in this repository than every other users who is welcomed to commend an give there opinion.

If you are trying to say something regarding the mypdns then the correct path is to reply to the right threads. I do not offering support for mypdns is other peoples project

- fireworkanalytics.com: https://mypdns.org/my-privacy-dns/matrix/-/issues/3299
- fireworkadservices.com: https://mypdns.org/my-privacy-dns/matrix/-/issues/3298

> .... You do not block the youtube video, right?

Of curse I am.... 

![image](https://user-images.githubusercontent.com/44526987/142619581-21cba94e-238d-4592-939e-35b2fa6cde42.png)

**Comment by ryanbr** [comment]
Closing after much discussion.  It's not like Youtube, or Vimeo. User content being made not specifically for advertising (products/websites/services/etc). Youtube is just user content embedded video, it's primary function. To call any invideo advertising "it's like youtube" misses the point, and its often used a defense by invideo providers not wanting to be blocked.

Not to say we haven't seen websites sites use youtube.com to embed a "video advert". Which we also blocked in Easylist. But that is an exception to the rule, very rarely have I see this occur recently. So rare I can't actually recall the last time this occurred.

Firework isn't the only invideo advertising service being blocked in Easylist (no doubt we'll add more over time). Invideo advertising can't be made "Easylist Compliant".  If we allow one service this would open the doors for others.
